src/python/grpcio_observability/grpc_observability/_observability.py

```diff
@@ -0,0 +1,175 @@
+from opencensus.trace import trace_options as trace_options_module
+
+_LOGGER = logging.getLogger(__name__)
+
+ClientCallTracerCapsule = Any  # it appears only once in the function signature
+ServerCallTracerFactoryCapsule = Any  # it appears only once in the function signature
+grpc_observability = Any  # grpc_observability.py imports this module.
+
+GRPC_STATUS_CODE_TO_STRING = {
+    grpc.StatusCode.OK: "OK",
+    grpc.StatusCode.CANCELLED: "CANCELLED",
+    grpc.StatusCode.UNKNOWN: "UNKNOWN",
+    grpc.StatusCode.INVALID_ARGUMENT: "INVALID_ARGUMENT",
+    grpc.StatusCode.DEADLINE_EXCEEDED: "DEADLINE_EXCEEDED",
+    grpc.StatusCode.NOT_FOUND: "NOT_FOUND",
+    grpc.StatusCode.ALREADY_EXISTS: "ALREADY_EXISTS",
+    grpc.StatusCode.PERMISSION_DENIED: "PERMISSION_DENIED",
+    grpc.StatusCode.UNAUTHENTICATED: "UNAUTHENTICATED",
+    grpc.StatusCode.RESOURCE_EXHAUSTED: "RESOURCE_EXHAUSTED",
+    grpc.StatusCode.FAILED_PRECONDITION: "FAILED_PRECONDITION",
+    grpc.StatusCode.ABORTED: "ABORTED",
+    grpc.StatusCode.OUT_OF_RANGE: "OUT_OF_RANGE",
+    grpc.StatusCode.UNIMPLEMENTED: "UNIMPLEMENTED",
+    grpc.StatusCode.INTERNAL: "INTERNAL",
+    grpc.StatusCode.UNAVAILABLE: "UNAVAILABLE",
+    grpc.StatusCode.DATA_LOSS: "DATA_LOSS",
+}
+
+
+@dataclass
+class GcpObservabilityPythonConfig:
+    _singleton = None
+    _lock: threading.RLock = threading.RLock()
+    project_id: str = ""
+    stats_enabled: bool = False
+    tracing_enabled: bool = False
+    labels: Optional[Mapping[str, str]] = field(default_factory=dict)
+    sampling_rate: Optional[float] = 0.0
+
+    @staticmethod
+    def get():
+        with GcpObservabilityPythonConfig._lock:
+            if GcpObservabilityPythonConfig._singleton is None:
+                GcpObservabilityPythonConfig._singleton = GcpObservabilityPythonConfig(
+                )
+        return GcpObservabilityPythonConfig._singleton
+
+    def set_configuration(self,
+                          project_id: str,
+                          sampling_rate: Optional[float] = 0.0,
+                          labels: Optional[Mapping[str, str]] = None,
+                          tracing_enabled: bool = False,
+                          stats_enabled: bool = False) -> None:
+        self.project_id = project_id
+        self.stats_enabled = stats_enabled
+        self.tracing_enabled = tracing_enabled
+        self.labels = labels
+        self.sampling_rate = sampling_rate
+
+
+# pylint: disable=no-self-use
+class GCPOpenCensusObservability(grpc._observability.ObservabilityPlugin):
+    config: GcpObservabilityPythonConfig
+    exporter: "grpc_observability.Exporter"
+
+    def __init__(self, exporter: "grpc_observability.Exporter" = None):
+        self.exporter = None
+        self.config = GcpObservabilityPythonConfig.get()
+        if exporter:
+            self.exporter = exporter
+        else:
+            pass
+            # Actual implementation of OC exporter
```

Comment:
So there isn't a default exporter?

Reply:
No need to implement a mock now. Just raise a [`NotImplementedError`](https://docs.python.org/3/library/exceptions.html#NotImplementedError) here and add a TODO pointing to a Github issue to take care of this.